pkg/virt-handler/vm.go

```diff
@@ -2015,7 +2018,10 @@ func (d *VirtualMachineController) processVmUpdate(origVMI *v1.VirtualMachineIns
 			d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is migrating.")
 		}
 	} else {
-
+		// Umount any disks no longer mounted
+		if err := d.hotplugVolumeMounter.Unmount(vmi); err != nil {
+			return err
+		}
```

Comment:
This handles dynamically removing volumes at runtime, but won't handle every situation of removing volumes after shutdown. 

CLeaning up vmi's after deletion has been something we struggled with for years. We've finally got it mostly sorted at this point. Put a new function (maybe called something like "UnmountAll()") which searches through the local cache using a vmi's UID and unmounts all attachment pods into the ```processVmCleanup()``` function.  Anything in processVmCleanup() is guaranteed to be successfully executed after the VMI is deleted no matter what.